### DISCUSSION TITLE
支持http接口触发命令执行，可供外部平台对接

### DISCUSSION THREAD

**Discussion by Teingi** [oceanbase/obdiag#37] [Ideas]
目前的黑屏命令行方式对于第三方平台对接不够方便，希望能支持http接口触发命令执行

**Comment by wayyoungboy** [comment]
由于一般诊断工具的任务执行时间较长，需要将接口设计为异步化。由于异步化，因此需要维护每次指定任务的状态和对应的tranid。同时，需要考虑兼容目前的不同功能的返回值类型，例如结果报告类、结果异常的设定等。同时http接口需要考虑后续的对接白屏化接口的介入。

Since the task execution time of general diagnostic tools is long, the interface needs to be designed to be asynchronous. Due to asynchronous, it is necessary to maintain the status and corresponding tranid of each specified task. At the same time, it is necessary to consider return value types that are compatible with different current functions, such as result reporting classes, result exception settings, etc. At the same time, the http interface needs to consider the subsequent intervention of the white screen interface.
